### PR TITLE
virttest.qemu_vm: Add OVMF command line in qemu_vm

### PR DESCRIPTION
**Pull request by ypu**

### CODE REVIEW

**Comment by ypu** [comment]
ID: 1208727

**Comment by ypu** [comment]
@suqinhuang please install the OVMF package first. If you didn't install it the code will record warning and skip add the device

**Comment by suqinhuang** [comment]
1. Still no ovmf in cmd
[root@dhcp-10-214 ~]# rpm -qa | grep -i ovmf
OVMF-20150414-1.gitc9e5618.el7.noarch
[root@dhcp-10-214 ~]# rpm -ql OVMF-20150414-1.gitc9e5618.el7.noarch
/usr/share/OVMF
/usr/share/OVMF/OVMF_CODE.fd
/usr/share/OVMF/OVMF_VARS.fd

2. Suggest to fail the test instead of thread warning when ovmf is not installed

**Comment by suqinhuang** [comment]
/bin/qemu-kvm \
    -S  \
    -name 'virt-tests-vm1'  \
    -sandbox off  \
    -M pc  \
    -nodefaults  \
    -vga std  \
    -chardev socket,id=qmp_id_qmpmonitor1,path=/tmp/monitor-qmpmonitor1-20150529-113243-bLbNIwJn,server,nowait \
    -mon chardev=qmp_id_qmpmonitor1,mode=control  \
    -chardev socket,id=serial_id_serial0,path=/tmp/serial-serial0-20150529-113243-bLbNIwJn,server,nowait \
    -device isa-serial,chardev=serial_id_serial0  \
    -chardev socket,id=seabioslog_id_20150529-113243-bLbNIwJn,path=/tmp/seabios-20150529-113243-bLbNIwJn,server,nowait \
    -device isa-debugcon,chardev=seabioslog_id_20150529-113243-bLbNIwJn,iobase=0x402 \
    -device ich9-usb-uhci1,id=usb1,bus=pci.0,addr=03 \
    -drive id=drive_image1,if=none,cache=none,snapshot=off,aio=native,file=/root/patch_review/autotest-devel/client/tests/virt/shared/data/images/win7-64-sp1-virtio.qcow2 \
    -device virtio-blk-pci,id=image1,drive=drive_image1,bootindex=0,bus=pci.0,addr=04 \
    -device virtio-net-pci,mac=9a:be:bf:c0:c1:c2,id=id8uOmuk,vectors=4,netdev=id4B4Cej,bus=pci.0,addr=05  \
    -netdev tap,id=id4B4Cej,vhost=on,vhostfd=23,fd=22  \
    -m 4096  \
    -smp 1,maxcpus=1,cores=1,threads=1,sockets=1  \
    -cpu 'Opteron_G2',hv_relaxed,+kvm_pv_unhalt,hv_relaxed,hv_spinlocks=0x1fff,hv_vapic,hv_time \
    -drive id=drive_cd1,if=none,snapshot=off,aio=native,media=cdrom,file=/root/patch_review/autotest-devel/client/tests/virt/shared/data/isos/ISO/Windows7/en_windows_7_ultimate_with_sp1_x64_dvd_u_677332.iso \
    -device ide-cd,id=cd1,drive=drive_cd1,bootindex=1,bus=ide.0,unit=0 \
    -drive id=drive_winutils,if=none,snapshot=off,aio=native,media=cdrom,file=/root/patch_review/autotest-devel/client/tests/virt/shared/data/isos/windows/winutils.iso \
    -device ide-cd,id=winutils,drive=drive_winutils,bootindex=2,bus=ide.0,unit=1 \
    -drive id=drive_unattended,if=none,snapshot=off,aio=native,media=cdrom,file=/root/patch_review/autotest-devel/client/tests/virt/shared/data/images/win7-64-sp1/autounattend.iso \
    -device ide-cd,id=unattended,drive=drive_unattended,bootindex=3,bus=ide.1,unit=0 \
    -device usb-tablet,id=usb-tablet1,bus=usb1.0,port=1  \
    -vnc :0  \
    -rtc base=localtime,clock=host,driftfix=slew  \
    -boot order=cdn,once=d,menu=off,strict=off \
    -enable-kvm

**Comment by ypu** [comment]
Thanks @suqinhuang for your comment. Already update the pullreq please help to review it again.

File: virttest/qemu_vm.py [hunk 1]
```diff
@@ -1852,6 +1853,35 @@ def add_balloon(devices, devid=None, bus=None, use_old_format=None):
         if bios_path:
             devices.insert(StrDev('bios', cmdline="-bios %s" % bios_path))
 
+        if params.get('ovmf_path'):
+            if not os.path.exists(params['ovmf_path']):
+                raise error.TestError("The OVMF path is not exist. Maybe you"
+                                      " need to install related packages.")
+            autotest_data_dir = data_dir.get_data_dir()
+            ovmf_code_path = os.path.join(params['ovmf_path'],
+                                          "OVMF_CODE.fd")
```
Comment (suqinhuang) [inline virttest/qemu_vm.py]: Hi ypu,

How about use the path of the file directly, need to update the code if the file name is updated one day.

Comment (ypu) [inline virttest/qemu_vm.py]: OK. any other please need update?

**Comment by suqinhuang** [comment]
Acked-by: Suqin Huang

**Comment by ypu** [comment]
Thanks @suqinhuang and @spcui for your review. Will merge it.

**Pull request merged**